### PR TITLE
yj: 4.0.0 -> 5.0.0

### PR DESCRIPTION
**Pull request by nightmared**
###### Motivation for this change

This PR is necessary to advance https://github.com/cargo2nix/cargo2nix/pull/165
(older versions of yj didn't differenciate between integers and floats,
which can break cargo manifests when converting from TOML to JSON and back).

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/tools/yj/default.nix [hunk 1]
```diff
@@ -1,20 +1,20 @@
...
-{ stdenv, buildGoPackage, fetchgit, fetchhg, fetchbzr, fetchsvn }:
+{ stdenv, buildGoModule, fetchFromGitHub }:
 
-buildGoPackage rec {
+buildGoModule rec {
   pname = "yj";
-  version = "4.0.0";
-  rev = "d9a48607cc5c812e8cf4abccc8ad26f37ab51558";
+  version = "5.0.0";
+  rev = "c4c13b7641389c76ea028b48091f851f3efb6376";
```
Comment (SuperSandro2000) [inline pkgs/development/tools/yj/default.nix:6]:
```suggestion
```

File: pkgs/development/tools/yj/default.nix [hunk 2]
```diff
@@ -1,20 +1,20 @@
...
 
-  goPackagePath = "github.com/sclevine/yj";
-
-  src = fetchgit {
+  src = fetchFromGitHub {
+    owner = "sclevine";
+    repo = "yj";
     inherit rev;
```
Comment (SuperSandro2000) [inline pkgs/development/tools/yj/default.nix:11]:
```suggestion
    rev = "c4c13b7641389c76ea028b48091f851f3efb6376";
```

File: pkgs/development/tools/yj/default.nix [hunk 3]
```diff
@@ -1,20 +1,20 @@
...
-    url = "https://github.com/sclevine/yj";
-    sha256 = "04irphzs6hp9hvyski29ad29ga1kis3h8bw7jqvmy2c2rkrrsh7x";
+    sha256 = "0bnb88wfm2vagh4yb1h9xhp3045ga0b6a77n3j2z5b4mvwshx5dr";
   };
 
-  goDeps = ./deps.nix;
+  vendorSha256 = "0y0n9fsb85qlpf9slwsxzarmfi98asa4x04qp2r8pagl28l0i8wv";
+
+  buildFlagsArray = [ "-ldflags=-X main.Version=${version}" ];
```
Comment (SuperSandro2000) [inline pkgs/development/tools/yj/default.nix:17]:
```suggestion
  buildFlagsArray = [ "-ldflags=-s -w -X main.Version=${version}" ];
```

File: pkgs/development/tools/yj/default.nix [hunk 4]
```diff
@@ -1,20 +1,20 @@
...
 
   meta = with stdenv.lib; {
     description = ''Convert YAML <=> TOML <=> JSON <=> HCL'';
```
Comment (SuperSandro2000) [inline pkgs/development/tools/yj/default.nix:20]: Why inherit version here? Did not see that before.

Comment (nightmared) [reply pkgs/development/tools/yj/default.nix:20]: When one calls `yj -v` to see the version of yj installed, `yj` prints the main.Version variable, that defaults to '0.0.0', so the real number is actually supplied at compile time. It also appears to be the way chosen by the developer of the package: https://github.com/sclevine/yj/search?q=Version

Comment (SuperSandro2000) [reply pkgs/development/tools/yj/default.nix:19]: ``buildFlagsArray`` is fine. I mean the now removed line 25.

**Comment by SuperSandro2000** [comment]
Result of `nixpkgs-review pr 107484` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>poetry</li>
    <li>yj</li>
  </ul>
</details>

**Comment by SuperSandro2000** [comment]
Result of `nixpkgs-review pr 107484` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>poetry</li>
    <li>yj</li>
  </ul>
</details>

**Pull request merged**